python/whylogs/core/declarative_schema.py

```diff
@@ -0,0 +1,84 @@
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Mapping, Optional, TypeVar
+
+from whylogs.core.datatypes import DataType, TypeMapper
+from whylogs.core.metrics.metrics import Metric, MetricConfig
+from whylogs.core.resolvers import Resolver
+from whylogs.core.schema import DatasetSchema
+from whylogs.core.segmentation_partition import SegmentationPartition
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class MetricSpec:
+    metric: Any  # Should be a Metric class
+    config: Optional[MetricConfig] = None
+
+
+@dataclass
+class ResolverSpec:
+    column_name: Optional[str] = None  # TODO: maybe make this a regex
+    column_type: Optional[DataType] = None
+    metrics: List[MetricSpec] = field(default_factory=list)
+
+
+class DeclarativeResolver(Resolver):
+
+    def __init__(
+            self,
+            resolvers: List[ResolverSpec],
+            default_config: Optional[MetricConfig]
```

Comment:
is `default_config` actually being used?

Reply:
It gets used via `ColumnSchema::cfg` if the `MetricSpec::config` doesn't override it.